go-server-server/go/default.go

```diff
@@ -1051,62 +1055,87 @@ func ConfigVrouterVrfIdRoutesPatch(w http.ResponseWriter, r *http.Request) {
         return
     }
 
-    pt := swsscommon.NewProducerStateTable(db.swss_db, ROUTE_TUN_TB)
+    var pt swsscommon.ProducerStateTable
+    var rt_tb_name string
     defer pt.Delete()
 
     var failed []RouteModel
 
     for _, r := range attr {
+        if r.IfName == "" {
+            pt = swsscommon.NewProducerStateTable(db.swss_db, ROUTE_TUN_TB)
+            rt_tb_name = ROUTE_TUN_TB
+            if *RunApiAsLocalTestDocker {
+                rt_tb_name = "_"+ROUTE_TUN_TB
+            }            
+        } else {
+            pt = swsscommon.NewProducerStateTable(db.swss_db, LOCAL_ROUTE_TB)
```

Comment:
Curious if there is any perf impact to creating NewProducerStateTable objects like this in a loop vs earlier model where we just do a single NewProducerStateTable and use it across the entire set of routes. Considering 100s of routes being programmed in a single call.

Reply:
Good point! i think we can have the pt allocation outside and just assign it inside the loop. What you guys think?

Reply:
I think that makes sense, would be good to do it that way.